pallets/pallet-bonded-coins/src/tests/transactions/finish_destroy.rs

```diff
@@ -0,0 +1,285 @@
+				.expect("failed to set up test state: asset cannot be set to destroying");
+
+			let origin = RawOrigin::Signed(ACCOUNT_00).into();
+
+			assert_ok!(BondingPallet::finish_destroy(origin, pool_id.clone(), 1));
+
+			// Verify that the pool state entry has been removed
+			assert_eq!(Pools::<Test>::get(&pool_id), None);
+
+			// Verify the expected event has been deposited
+			System::assert_has_event(Event::Destroyed { id: pool_id }.into());
+		});
+}
+
+#[test]
+fn fails_on_incorrect_state() {
+	let pool_details = generate_pool_details(
+		vec![DEFAULT_BONDED_CURRENCY_ID],
+		get_linear_bonding_curve(),
+		true,
+		Some(PoolStatus::Active),
+		None,
+		Some(DEFAULT_COLLATERAL_CURRENCY_ID),
+		Some(ACCOUNT_00), // owner must hold native asset so we can reserve deposit
+	);
+	let pool_id = calculate_pool_id(&[DEFAULT_BONDED_CURRENCY_ID]);
+	let origin: OriginFor<Test> = RawOrigin::Signed(ACCOUNT_00).into();
+
+	ExtBuilder::default()
+		.with_pools(vec![(pool_id.clone(), pool_details.clone())])
+		.with_native_balances(vec![(ACCOUNT_00, ONE_HUNDRED_KILT)])
+		.with_collaterals(vec![DEFAULT_COLLATERAL_CURRENCY_ID])
+		.build()
+		.execute_with(|| {
+			// Assets need to be in destroying state if pool is in destroying state
+			<<Test as crate::Config>::Fungibles as Destroy<_>>::start_destroy(DEFAULT_BONDED_CURRENCY_ID, None)
+				.expect("failed to set up test state: asset cannot be set to destroying");
+
+			assert_err!(
+				BondingPallet::finish_destroy(origin.clone(), pool_id.clone(), 1),
+				Error::<Test>::LivePool
+			);
+
+			Pools::<Test>::mutate(&pool_id, |details| {
+				details.as_mut().unwrap().state.start_refund();
+			});
+
+			assert_err!(
+				BondingPallet::finish_destroy(origin.clone(), pool_id.clone(), 1),
+				Error::<Test>::LivePool
+			);
+
+			Pools::<Test>::mutate(&pool_id, |details| {
+				details.as_mut().unwrap().state.start_destroy();
+			});
+
+			assert_ok!(BondingPallet::finish_destroy(origin, pool_id.clone(), 1));
+		});
+}
+
+#[test]
+fn fails_if_assets_cannot_be_destroyed() {
+	let pool_details = generate_pool_details(
+		vec![DEFAULT_BONDED_CURRENCY_ID],
+		get_linear_bonding_curve(),
+		true,
+		Some(PoolStatus::Destroying),
+		None,
+		Some(DEFAULT_COLLATERAL_CURRENCY_ID),
+		Some(ACCOUNT_00), // owner must hold native asset so we can reserve deposit
+	);
+	let pool_id = calculate_pool_id(&[DEFAULT_BONDED_CURRENCY_ID]);
+	let origin: OriginFor<Test> = RawOrigin::Signed(ACCOUNT_00).into();
+
+	ExtBuilder::default()
+		.with_pools(vec![(pool_id.clone(), pool_details.clone())])
+		.with_native_balances(vec![(ACCOUNT_00, ONE_HUNDRED_KILT)])
+		.with_collaterals(vec![DEFAULT_COLLATERAL_CURRENCY_ID])
+		.with_bonded_balance(vec![(DEFAULT_BONDED_CURRENCY_ID, ACCOUNT_00, 100_000)])
+		.build()
+		.execute_with(|| {
+			// Fails because asset is not in destroying state
+			assert!(BondingPallet::finish_destroy(origin.clone(), pool_id.clone(), 1).is_err());
```

Comment:
`assert_eq(BondingPallet::.finish_destroy(origin.clone(), pool_id.clone(), 1), Err(_));` is not possible?

Reply:
I think we have to use assert_eq. We introduced a neew lint, to avoid blank assert on results.

Reply:
I think assert_eq only works for tests for equality, which don't allow matching patterns afaik. Same for assert_err!, which is why I'm not using that here. We can use a `matches!` expression, but the result is exactly the same as using the `is_err()`.

Reply:
I don't know, which one do you prefer?

Reply:
Then leave it like that :+1: